doc/architecture/decisions/0002-pathogens-portal-2-0-architecture.md

```diff
@@ -0,0 +1,150 @@
+
+* **External Services**: These services will be consumed by the backend via their APIs:
+
+  * **Portal storage solution:** for data uploads and file access
+  * **Integrated Data Services (IDS):** for harmonised metadata and research objects
+  * **External repositories:** e.g. publication indexes, genomic data archives
+
+### Why This Architecture Approach?
+
+We chose a **service-oriented architecture (SOA)** because it provides:
+
+* **Loose coupling:** Components can evolve independently, enabling isolated deployments and targeted updates without affecting other parts of the application.
+* **Improved scalability:** Each service can scale based on its own resource needs, supporting future growth in data volume and user traffic.
+* **Separation of concerns:** Clean boundaries between UI, logic, and storage layers, facilitating easier maintenance and feature development.
+* **Better integration:** External APIs can be consumed easily via backend orchestration or direct frontend queries, enabling seamless data pipeline integration.
+* **AI/ML readiness:** Modular design supports easy integration of AI services, machine learning pipelines, and automated data processing workflows.
+* **Future feature flexibility:** Architecture supports rapid prototyping and deployment of new features like real-time data visualisation, collaborative tools, and advanced analytics.
+* **Enhanced data pipeline integration:** Service boundaries allow for sophisticated ETL processes, data validation, and transformation workflows without impacting user-facing components.
+* **Microservices evolution path:** Current SOA design provides a foundation for future migration to microservices if needed for advanced use cases.
+* **API-first approach:** Enables multiple client types (web, mobile, third-party integrations) while maintaining consistent data access patterns.
+
+This approach aligns with modern web development standards preparing the portal for more complex workflows and higher data integration.
+
+
+### Why These Specific Technologies?
+
+| Component    | Technology | Justification                                                                                                                                                                     |
+| ------------ | ---------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Frontend** | Next.js    | Combines static and dynamic rendering, SEO-friendly, React-based, and widely supported. Flexible enough to handle dashboard-style applications and traditional content rendering. |
+| **Backend**  | Django     | Secure, robust, and feature-rich with a built-in admin interface, powerful ORM, and mature REST tooling. Well known to the team and easy to maintain.                             |
```

Comment:
If you are going Next.JS route, are you going to use pure Django, DRF or Django Ninja?

If it's not decided yet, I can recommend taking a look at django ninja

Reply:
We haven't talked about this yet, but off the top of my head I was indeed only considering Ninja or DRF if we go the full API route. I'll read more in depth about Ninja, thanks for the hint ;)